### ISSUE TITLE
"Cannot find name 'unknown'" despite having installed unknown-ts

### ISSUE THREAD

**Issue by jjoekoullas** [gcanti/fp-ts#724]
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behavior?**
Using typescript 2.4.0, with unknown-ts & fp-ts installed, I'm getting compilation errors:

> node_modules/fp-ts/lib/Either.d.ts(275,35): error TS2304: Cannot find name 'unknown'

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://codesandbox.io/ or similar.**
Minimal reproduction of the issue can be found here: https://github.com/jjoekoullas/missing-unknown/commit/5e67dbf304db279e3e6cdefb621fb050ae823c9d

**What is the expected behavior?**
No errors (or stated incompatibility with earlier versions of typescript outside of workaround)

**Which versions of fp-ts, and which browser and OS are affected by this issue? Did this work in previous versions of fp-ts?**
fp-ts: 1.13.0
unknown-ts: 0.1.0

Unsure about earlier versions

**Comment by sledorze** [comment]
@jjoekoullas not sure the project maintains that much legacy typescript versions.

The rational is:
Typescript type system is a very moving target with lot of breaking changes at each release.
Maintaining a lot of versions is close to impossible with the amount of type system features `fp-ts` is using.

@gcanti could state on that.

**Comment by jjoekoullas** [comment]
There's a comment close to the first line of the readme that says to use unknown-ts for backwards compatibility. If 2.4 is too backwards, that's fine, I think updating the docs is enough.

That said, I'd like to look into this and see if there is an easy fix. I looked through the existing issues and didn't see anybody else running into this, so I decided it was worth logging at least in case somebody else runs into the same issue.

"Just update TypeScript" unfortunately isn't a good option for my organization. I truly wish it was.

All the best
